File: cuaderno_mantenimiento_backend/src/intervention/dto/create-intervention.dto.ts
import { IsUUID, IsDateString, IsNumber, IsString, IsOptional, IsNotEmpty, IsIn } from 'class-validator';

export class CreateInterventionDto {
  @IsNotEmpty()
  @IsDateString()
  fecha: string;

  @IsNotEmpty()
  @IsNumber()
  kilometraje: number;

  @IsNotEmpty()
  @IsString()
  @IsIn(['Revision', 'Cambio de pieza', 'Reparacion'], {
    message: 'tipo_intervencion must be either "Revision", "Cambio de pieza" or "Reparacion"',
  })
  tipo_intervencion: string;

  @IsOptional()
  @IsString()
  observaciones?: string;

  @IsNotEmpty()
  @IsUUID()
  vehicle_id: string; // Relación con Vehicle
}

File: cuaderno_mantenimiento_flutter/lib/infrastructure/dtos/create-intervention-dto.dart
class CreateInterventionDto {
  final String fecha;
  final int kilometraje;
  final String tipoIntervencion;
  final String? observaciones;
  final String vehicle_id;

  CreateInterventionDto({
    required this.fecha,
    required this.kilometraje,
    required this.tipoIntervencion,
    this.observaciones,
    required this.vehicle_id,
  });

  Map<String, dynamic> toJson() {
    return {
      'fecha': fecha,
      'kilometraje': kilometraje,
      'tipo_intervencion': tipoIntervencion,
      'observaciones': observaciones,
      'vehicle_id': vehicle_id,
    };
  }
}
